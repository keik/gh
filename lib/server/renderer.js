// @flow

/**
 * Return full HTML string with React props
 *
 * @param {string} content content as HTML string
 * @param {object} props React props object for Server-Side Rendering
 * @param {string} entryName name of entrypoint
 */
export function renderFullPage(
  content: string,
  props: any,
  entryName: string,
  bust: number | string
) {
  return `
 <!DOCTYPE html>
 <html>
   <head>
     <meta charset="utf-8">
     <title>gh</title>
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css">
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.7.0/styles/github.min.css">
     <link href="https://fonts.googleapis.com/css?family=Rajdhani" rel="stylesheet" />
     <link rel="stylesheet" href="/style.css?${bust}">
   </head>
   <body>
     <div id="app">${content}</div>
     <script>APP_PROPS = ${JSON.stringify(props)}</script>
     <script src="/${entryName}.js?${bust}"></script>
   </body>
 </html>
 `
}
