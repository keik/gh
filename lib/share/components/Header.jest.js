// @flow

import * as React from 'react'
import renderer from 'react-test-renderer'

import Header from './Header'

test('<Header /> snapshot', () => {
  const tree = renderer.create(<Header session={undefined} />).toJSON()
  expect(tree).toMatchSnapshot()
})
