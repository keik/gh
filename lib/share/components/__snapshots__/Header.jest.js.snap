// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot 1`] = `
<div
  className="Header-without-session__Container-sc-1ortxq7-0 ghLqSv"
>
  <div>
    <a
      className="Logo-k763vu-0 eMaEfw"
      href="/"
    >
      GH
    </a>
  </div>
  <form
    className="Header-without-session__HeaderForm-sc-1ortxq7-1 hGFALm"
  >
    <label
      htmlFor="input-search-query"
    >
      This repository
    </label>
    <input
      id="input-search-query"
      placeholder="Search"
      type="text"
    />
  </form>
  <div
    className="ButtonsContainer-s9x8v3-0 hTAHWK"
  >
    <a
      className="Button-sc-1ldg16x-0 gIWfju"
      href="/login"
    >
      Sign in
    </a>
    <a
      className="Button-sc-1ldg16x-0 hZgZyG"
      href="/join"
    >
      Sign up
    </a>
  </div>
</div>
`;
